src/test/java/org/springframework/data/elasticsearch/core/sql/SqlOperationsIntegrationTests.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core.sql;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.springframework.data.elasticsearch.core.sql.types.ResponseFormat.yaml;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.elasticsearch.annotations.Document;
+import org.springframework.data.elasticsearch.core.ElasticsearchOperations;
+import org.springframework.data.elasticsearch.core.IndexOperations;
+import org.springframework.data.elasticsearch.core.query.SqlQuery;
+import org.springframework.data.elasticsearch.junit.jupiter.ElasticsearchTemplateConfiguration;
+import org.springframework.data.elasticsearch.junit.jupiter.SpringIntegrationTest;
+import org.springframework.lang.Nullable;
+import org.springframework.test.context.ContextConfiguration;
+
+/**
+ * Testing the querying using SQL syntax.
+ *
+ * @author Youssef Aouichaoui
+ */
+@SpringIntegrationTest
+@ContextConfiguration(classes = { SqlOperationsIntegrationTests.Config.class })
+@DisplayName("Using Elasticsearch SQL Client")
+class SqlOperationsIntegrationTests {
+	@Autowired ElasticsearchOperations operations;
+	@Nullable IndexOperations indexOps;
+
+	@BeforeEach
+	void setUp() {
+		// create index
+		indexOps = operations.indexOps(EntityForSQL.class);
+		indexOps.createWithMapping();
+
+		// add data
+		operations.save(EntityForSQL.builder().withViews(3).build(), EntityForSQL.builder().withViews(0).build());
+	}
+
+	@AfterEach
+	void tearDown() {
+		// delete index
+		if (indexOps != null) {
+			indexOps.delete();
+		}
+	}
+
+	// begin configuration region
+	@Configuration
+	@Import({ ElasticsearchTemplateConfiguration.class })
+	static class Config {}
+	// end region
+
+	@Test
```

Comment:
```suggestion
	@Test // #2683
```

please add the issue number to the tests, overlooked that on the first review